pubsub/topic.go

```diff
@@ -238,17 +247,21 @@ type TopicConfigToUpdate struct {
 
 	// If set to a positive duration between 10 minutes and 7 days, RetentionDuration is changed.
 	// If set to a negative value, this clears RetentionDuration from the topic.
-	RetentionDuration time.Duration
+	// If nil, the retention duration remains unchanged.
+	RetentionDuration optional.Duration
 }
 
 func protoToTopicConfig(pbt *pb.Topic) TopicConfig {
-	return TopicConfig{
+	tc := TopicConfig{
 		Labels:               pbt.Labels,
 		MessageStoragePolicy: protoToMessageStoragePolicy(pbt.MessageStoragePolicy),
 		KMSKeyName:           pbt.KmsKeyName,
 		SchemaSettings:       protoToSchemaSettings(pbt.SchemaSettings),
-		RetentionDuration:    pbt.MessageRetentionDuration.AsDuration(),
 	}
+	if pbt.MessageRetentionDuration != nil {
+		tc.RetentionDuration = pbt.MessageRetentionDuration.AsDuration()
```

Comment:
A super nit: It is best practice to use the accessor methods when _reading_ a proto field value i.e. `pbt.GetMessageRetentionDuration()`. For example, if a field is a `*string`, `GetFoo` would deref `if Foo != nil` or return `""` `if Foo == nil`.

Reply:
Done